src/nsj_rest_lib/service/service_base.py

```diff
@@ -773,75 +773,90 @@ def _group_by_override_data(self, dto_list):
+                )
+
+            # Monta o filtro IN para buscar todos os relacionados de uma vez
+            filters = {
+                list_field.related_entity_field: ",".join([str(key) for key in key_to_dtos])
+            }
 
-                    # Making filter to relation
-                    filters = {list_field.related_entity_field: relation_filter_value}
+            # Campos de particionamento: se existirem, só faz sentido se todos os DTOs tiverem o mesmo valor
+            # (caso contrário, teria que quebrar em vários queries)
+            # Aqui, só trata se todos tiverem o mesmo valor para cada campo de partição
+            for field in self._dto_class.partition_fields:
+                if field in list_field.dto_type.partition_fields:
+                    partition_values = set(getattr(dto, field, None) for dto in dto_list)
+                    partition_values.discard(None)
+                    if len(partition_values) == 1:
+                        filters[field] = partition_values.pop()
+                    # Se houver mais de um valor, teria que quebrar em vários queries (não tratado aqui)
+
+            # Resolvendo os fields da entidade aninhada
+            fields_to_list = copy.deepcopy(fields)
+            if master_dto_attr in fields:
+                fields_to_list["root"] = fields[master_dto_attr]
+                del fields_to_list[master_dto_attr]
+            else:
+                fields_to_list["root"] = set()
 
-                    # Tratando campos de particionamento
-                    for field in self._dto_class.partition_fields:
-                        if field in list_field.dto_type.partition_fields:
-                            filters[field] = getattr(dto, field)
+            # Busca todos os relacionados de uma vez
+            related_dto_list = service.list(
+                None, None, fields_to_list, None, filters
+            )
 
-                    # Resolvendo os fields da entidade aninhada
-                    fields_to_list = copy.deepcopy(fields)
-                    if master_dto_attr in fields:
-                        fields_to_list["root"] = fields[master_dto_attr]
-                        del fields_to_list[master_dto_attr]
-                    else:
-                        fields_to_list["root"] = set()
+            # Agrupa os relacionados por chave
+            related_map = {}
+            for related_dto in related_dto_list:
+                key = getattr(related_dto, list_field.related_entity_field, None)
+                if key is not None:
+                    related_map.setdefault(key, []).append(related_dto)
 
-                    # Getting related data
-                    related_dto_list = service.list(
-                        None, None, fields_to_list, None, filters
-                    )
+            # Seta nos DTOs principais
+            for key, dtos in key_to_dtos.items():
+                related = related_map.get(key, [])
```

Comment:
Não seria melhor um deepcopy aqui? Ou pelo menos a partir do segundo item do loop abaixo? Porque, os DTOs serão exatamente iguais, nos objetos que compartilharem a chave de relacionamento... Isso será raro. Mas, se algo for feito em um, será feito no outro... Novamente, esse cenário me parece estranho e raro...